test/cases/provider-id.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "--> Should fetch imds details correctly"
+EXPECTED_INSTANCE_ID="i-1234567890abcdef0"
+EXPECTED_AVAILABILITY_ZONE="us-east-1a"
+EXPECTED_PROVIDER_ID="aws://$EXPECTED_AVAILABILITY_ZONE/$EXPECTED_INSTANCE_ID"
+PROVIDER_ID=$(provider-id)
+if [ ! "$PROVIDER_ID" = "$EXPECTED_PROVIDER_ID" ]; then
+  echo "❌ Test Failed: expected provider-id=$EXPECTED_PROVIDER_ID but got '${PROVIDER_ID}'"
+  exit 1
+fi
+
+echo "--> Should fail when imds is unreachable"
+echo '#!/usr/bin/sh
+exit 1' > $(which imds)
```

Comment:
Will this get refreshed when other test cases run? I assume so.

Reply:
Also, per my comment above, is this the only way the imds call will fail? Want to make sure we don't miss out on any funkiness.

Reply:
We use a fresh container for each test case, yeah.

The `imds` script could fail for plenty of reasons, but would exit with a non-zero code. `set -o errexit` in `provider-id` would bail if `imds` croaks.

Reply:
>The imds script could fail for plenty of reasons, but would exit with a non-zero code. set -o errexit in provider-id would bail if imds croaks.

This wasn't true when I posted it, but it's true for the latest revision 😄